extensions/kubernetes-client/deployment-internal/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java

```diff
@@ -0,0 +1,415 @@
+                                    .orElseGet(() -> latest(K3sContainerVersion.class)));
+                    break;
+                case KIND:
+                    container = new KindContainer(
+                            config.apiVersion.map(version -> findOrElseThrow(version, KindContainerVersion.class))
+                                    .orElseGet(() -> latest(KindContainerVersion.class)));
+                    break;
+                default:
+                    throw new RuntimeException();
+            }
+
+            if (useSharedNetwork) {
+                ConfigureUtil.configureSharedNetwork(container, "quarkus-kubernetes-client");
+            }
+            if (config.serviceName != null) {
+                container.withLabel(DevServicesKubernetesProcessor.DEV_SERVICE_LABEL, config.serviceName);
+            }
+            timeout.ifPresent(container::withStartupTimeout);
+
+            container.start();
+
+            KubeConfig kubeConfig = KubeConfigUtils.parseKubeConfig(container.getKubeconfig());
+
+            return new RunningDevService(Feature.KUBERNETES_CLIENT.getName(), container.getContainerId(),
+                    new ContainerShutdownCloseable(container, Feature.KUBERNETES_CLIENT.getName()),
+                    getKubernetesClientConfigFromKubeConfig(kubeConfig));
+        };
+
+        return maybeContainerAddress
+                .map(containerAddress -> new RunningDevService(Feature.KUBERNETES_CLIENT.getName(),
+                        containerAddress.getId(),
+                        null,
+                        resolveConfigurationFromRunningContainer(containerAddress)))
+                .orElseGet(defaultKubernetesClusterSupplier);
+    }
+
+    <T extends KubernetesVersionEnum<T>> T findOrElseThrow(final String version, final Class<T> versions) {
+        final String versionWithPrefix = !version.startsWith("v") ? "v" + version : version;
+        return Stream.of(versions.getEnumConstants())
+                .filter(v -> v.descriptor().getKubernetesVersion().equalsIgnoreCase(versionWithPrefix))
+                .findFirst()
+                .orElseThrow();
+    }
+
+    private Map<String, String> getKubernetesClientConfigFromKubeConfig(KubeConfig kubeConfig) {
+
+        ClusterSpec cluster = kubeConfig.getClusters().get(0).getCluster();
+        UserSpec user = kubeConfig.getUsers().get(0).getUser();
+        return Map.of(
+                KUBERNETES_CLIENT_MASTER_URL, cluster.getServer(),
+                "quarkus.kubernetes-client.ca-cert-data",
+                cluster.getCertificateAuthorityData(),
+                "quarkus.kubernetes-client.client-cert-data",
+                user.getClientCertificateData(),
+                "quarkus.kubernetes-client.client-key-data", user.getClientKeyData(),
+                "quarkus.kubernetes-client.client-key-algo", Config.getKeyAlgorithm(null, user.getClientKeyData()),
+                "quarkus.kubernetes-client.namespace", "default");
```

Comment:
Are we always using the "default" namespace on purpose? This can be set from kubeconfig as well.

Reply:
Actually, the test container starts with only the default namespace. We may support additional namespace by adding additional devservices config. But it can be done in a second step, there is a lot that can be configured at startup like applying yaml resource.